testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs

```diff
@@ -190,7 +197,7 @@ impl LeaderBlockCommitFees {
         let value_per_transfer = payload.burn_fee / number_of_transfers;
         let sortition_fee = value_per_transfer * number_of_transfers;
         let spent_in_attempts = 0;
-        let fee_rate = config.burnchain.satoshis_per_byte;
+        let fee_rate = get_satoshis_per_byte(config);
```

Comment:
Because you're removing all instances of `config.burnchain.satoshis_per_byte`, can you take some preventative steps to make sure that `config.burnchain.satoshis_per_byte` cannot be accidentally accessed?  For example, is it possible to make `satoshis_per_byte` private?

Reply:
It's used as the backup in get_satoshis_per_byte, but there might be some way around it

Reply:
you _could_ load a default BurnchainConfig in the `Err` case of `get_satoshis_per_byte` instead of accessing the one in config.burnchain but it's not the same. I think it would be pretty weird to introduce a different config in the case it can't find the original, but it seems like such an edge case it might not matter?

I'm just thinking of a scenario where they run the miner with a spb of 100, then the config gets moved or deleted, get_satoshis_per_byte would look for it, then default to BurnchainConfig::default() and use 50, which is definitely not what they want

Reply:
> you _could_ load a default BurnchainConfig in the `Err` case of `get_satoshis_per_byte` instead of accessing the one in config.burnchain but it's not the same. I think it would be pretty weird to introduce a different config in the case it can't find the original, but it seems like such an edge case it might not matter?
> 
> I'm just thinking of a scenario where they run the miner with a spb of 100, then the config gets moved or deleted, get_satoshis_per_byte would look for it, then default to BurnchainConfig::default() and use 50, which is definitely not what they want

I don't think we need to worry about using a default burnchain config for a miner, since the miner will need to have a private key in order to broadcast btc txs. 
I would prefer an `Err` here if the config is moved or deleted.

Reply:
Yeah same

Reply:
Regarding my original question, I think what you have is fine.